pkg/graph/resolvers/system_intake_grb_reviewer.go

```diff
@@ -223,3 +235,17 @@ func StartGRBReview(
 		return helpers.PointerTo("started GRB review"), nil
 	})
 }
+
+func GetPrincipalAsGRBReviewerBySystemIntakeID(ctx context.Context, systemIntakeID uuid.UUID) (*models.SystemIntakeGRBReviewer, error) {
+	principalUserAcctID := appcontext.Principal(ctx).Account().ID
+	grbReviewers, err := dataloaders.GetSystemIntakeGRBReviewersBySystemIntakeID(ctx, systemIntakeID)
+	if err != nil {
+		return nil, err
+	}
+	for _, reviewer := range grbReviewers {
+		if reviewer != nil && reviewer.UserID == principalUserAcctID {
+			return reviewer, nil
+		}
+	}
+	return nil, nil
```

Comment:
if the caller receives `nil` for both return values, can that cause an issue? could it be better to `return nil, errors.New("unexpected missing principal as GRB reviewer")` or something similar?

Reply:
I think we should add some comments to this function explaining it's intended purpose. 

When we call this function, for example [here](https://github.com/CMS-Enterprise/easi-app/blob/9f8f43673e14cbce59c925bb61721e85aadfc6cf/pkg/graph/resolvers/system_intake_grb_discussions.go#L111-L120), we check `err != nil`, but later down the line also do some checks on if the reviewer is `nil`.

I think the case for `nil, nil` is basically "We had no unexpected errors calling the DB, but also the principal in the `ctx` is not a GRB reviewer" (which is basically to say that they're likely an admin)

tl;dr - I think `nil, nil` is still a valid return, but we should have comments on this function to explain that!